### DISCUSSION TITLE
Create API function to feed json directly to the parser

### DISCUSSION THREAD

**Discussion by leahoswald** [mealie-recipes/mealie#3224] [Feature Request]
### First Check

- [X] I used the GitHub search to find a similar requests and didn't find it.
- [X] Checked the [tasks tagged](https://github.com/mealie-recipes/mealie/issues?q=is%3Aissue+is%3Aopen+label%3Atask+) issues and verified my feature is not covered

### Please provide a concise description of the problem that would be addressed by this feature.

Some sites can not be parsed properly by the URL parser feature. For example because the ld+json is assembled in the browser with javascript. Copying the recipe manually is annoying and can be automated by a parser using selenium and a real browser. This parser could run locally so there is no need for a full browser on my server.  The problem is, now I have the json/html but can't feed it to the parser of mealie to extract all needed stuff and download images and so on. Creating it with the current API functions with a POST to `"base_url+"/api/recipes` and a PUT to `"base_url+"/api/recipes/{slug}` needs a lot of work to bring the json into a format that works with mealie, for example because of recipeCategory not beeing a list. 

### Please provide a concise description of the feature that would resolve your issue.

An API function like "Create Recipe From Zip" but you send json or html (easier) that gets feed to the same code like if I provide a URL to "Parse Recipe Url"

### Please consider and list out some caveats or tradeoffs made in your design decision

Should really be an easy fix.

### Additional Information

- [ ] If this is accepted I'm willing to submit a PR to provide this feature
- [ ] If this is accepted I'm willing to help maintain this feature
- [ ] I'm willing to sponsor/pay a developer to do this work

**Comment by michael-genson** [comment]
FYI this is now a feature: https://github.com/mealie-recipes/mealie/pull/4274

You can manually paste HTML or JSON (the JSON has to be in https://schema.org/Recipe format)

**Reply by leahoswald** [reply]
Wohoooo thanks!
